app/v2/controllers/V2MovementsController.scala

```diff
@@ -174,8 +186,20 @@ class V2MovementsControllerImpl @Inject() (
       implicit request => size =>
         implicit val hc: HeaderCarrier = HeaderCarrierConverter.fromRequest(request)
         (for {
-          _               <- contentSizeIsLessThanLimit(size)
-          _               <- validationService.validateXml(MessageType.DeclarationData, request.body).asPresentation
+          _ <- contentSizeIsLessThanLimit(size)
+          _ <- validationService.validateXml(MessageType.DeclarationData, request.body).asPresentation.leftMap {
```

Comment:
I think we also need to add ```ValidationFailed``` for validateJson method?

Reply:
Yes, we do.